crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,360 @@
+use alloc::{collections::BTreeMap, sync::Arc, vec::Vec};
+
+use miden_objects::{
+    account::{Account, AccountCode, AccountId, AccountProcedureInfo, AccountType},
+    assembly::mast::{MastForest, MastNode, MastNodeId},
+    crypto::dsa::rpo_falcon512,
+    note::{Note, NoteScript},
+    Digest,
+};
+
+use crate::{
+    account::components::{
+        basic_fungible_faucet_library, basic_wallet_library, rpo_falcon_512_library,
+    },
+    note::well_known_note::WellKnownNote,
+    AuthScheme,
+};
+
+#[cfg(test)]
+mod test;
+
+// ACCOUNT INTERFACE
+// ================================================================================================
+
+/// An [`AccountInterface`] describes the exported, callable procedures of an account.
+///
+/// A note script's compatibility with this interface can be inspected to check whether the note may
+/// result in a successful execution against this account.
+pub struct AccountInterface {
+    account_id: AccountId,
+    auth: Vec<AuthScheme>,
+    interfaces: Vec<AccountComponentInterface>,
```

Comment:
nit: actually, looking at this now, I think the more appropriate name is `components` rather than `interfaces`.